src/render_cmd.rs

```diff
@@ -151,8 +152,16 @@ impl RenderCmd {
 
             let mut cmd = base_cmd.clone();
 
+            // check if "--filter is set and filter the deployment values."
+            // return type will be a vector
+
+            let filtered_values = match &self.opts.filter {
+                Some(opts) => self.filter_deployment(opts.clone(), &d.name, d.values.clone()),
+                None => d.values.clone(),
+            };
+
             let values: Vec<String> = self
-                .get_values_as_strings(&d.values)?
+                .get_values_as_strings(&filtered_values)?
```

Comment:
I think there is no need to filter the values as `--filter` should only look at the deployment name.